docs/rancher.md

```diff
@@ -0,0 +1,34 @@
+# Deploying Antrea on Rancher Managed Cluster
+
+<!-- toc -->
+- [Deploy Rancher Server](#deploy-rancher-server)
+- [Deploy Antrea Using Rancher UI](#deploy-antrea-using-rancher-ui)
+<!-- /toc -->
```

Comment:
I feel like including a TOC for such a small document is not necessary. You could add the file to `hack/.notableofcontents` instead.